src/framework/mpas_timekeeping.F

```diff
@@ -1606,7 +1651,25 @@ subroutine mpas_get_timeInterval(interval, StartTimeIn, DD, H, M, S, S_n, S_d, S
       end if
 
       if (present(timeString)) then
+#ifndef MPAS_EXTERNAL_ESMF_LIB
          call ESMF_TimeIntervalGet(interval % ti, timeString=timeString, rc=ierr)
+#else
+         !
+         ! In case an external ESMF library is being used, the time interval may be returned in
+         ! ISO period format, in which case it is easier to build a time interval string in MPAS
+         ! format given days, hours, minutes, and seconds.
+         !
+         call ESMF_TimeIntervalGet(interval % ti, StartTimeIn=StartTimeIn%t, &
+                                   d=days, h=hours, m=minutes, s_i8=seconds, sN=sn, sD=sd, rc=ierr)
+         seconds_real = real(seconds,kind=R8KIND) + (real(sn, kind=R8KIND) / real(sd, kind=R8KIND))
+
+         ! Seconds should not be negative, so there is no need to handle negative cases, below
+         if (seconds_real < 10.0_R8KIND) then
+            write(timeString,'(i9.9,a,i2.2,a,i2.2,a,f11.9)') days, '_', hours, ':', minutes, ':0', seconds_real
+         else
+            write(timeString,'(i9.9,a,i2.2,a,i2.2,a,f12.9)') days, '_', hours, ':', minutes, ':', seconds_real
+         end if
```

Comment:
Just to be sure I understand whats going on here, the above if statement is being used to determine if the seconds in timeString should be padded with a leading 0 or not?

Reply:
That's correct.

Reply:
I don't know of a standard way to guarantee padding with leading zeros in Fortran, so if anyone has suggestions, here, I'd be glad to hear them.

Reply:
I think this works:
write(timeString,'(i9.9,a,i2.2,a,i2.2,a,i2.2,f10.9)') days, '_', hours, ':', minutes, ':', int(seconds_real),seconds_real-int(seconds_real)

Reply:
That seems to work well -- thanks for the suggestion! I'll add a new commit later this afternoon that takes this approach rather than having to select between two `write` statements depending on the magnitude of `seconds_real`.

Reply:
I've just pushed a new commit, 1d82840, that makes use of this, in addition to correcting an issue with negative time intervals.

Reply:
Looks good.